libs/generic-view/ui/src/lib/interactive/form/input/select-input.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { FormSelectInputConfig, FormSelectInputData } from "generic-view/models"
+import { APIFC, IconType } from "generic-view/utils"
+import { Icon } from "../../../icon/icon"
+import React, { useEffect, useRef, useState } from "react"
+import styled, { css } from "styled-components"
+
+export const SelectInput: APIFC<FormSelectInputData, FormSelectInputConfig> = ({
+  data,
+  config,
+  ...props
+}) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const selectRef = useRef<HTMLDivElement>(null)
+
+  const handleSelect = (selectedValue: string) => {
+    data!.value = selectedValue
```

Comment:
Ensure that `data` and `data.value` are defined before assignment to avoid runtime errors 🙏